src/datastructure.c

```diff
@@ -40,6 +40,17 @@ void strtolower(char *str)
 	while(str[i]){ str[i] = tolower(str[i]); i++; }
 }
 
+// creates a simple hash of a string that fits into a uint32_t
+uint32_t hashStr(const char *s)
+{
+        // Rotating hash (http://www.burtleburtle.net/bob/hash/doobs.html)
```

Comment:
Did you test in the other script that the simpler rotation hash also didn't cause any collisions in the Alexa 1 Mio list? I recall this was the case for the other hash we tried.

Reply:
I had actually forgot about the other hash you suggested in the test code, and have just done some additional testing on the collision rate (using the newer umbrella 1M domains that actually contain a million domains in the list).

```
Total collisions (rotating hash) 10356 in 1000000 domains
Total collisions (one at a time hash) 248 in 1000000 domains
```

That's quite a few less collisions!  I had assumed (incorrectly) that a 2**32 hash would have suffered from many collisions regardless and the simpler one would just save on some instructions.  Reading more closely I can see it never actually expands out to the whole effective space entirely.  The difference is pretty substantial and I will commit the one-at-a-time hash function in shortly after a little more testing on my pi.  

With this change I've managed to get my measured average cached dns response down to ~67ms now, for what it's worth, which is less than the ~80ms under the previous approaches (and ~100ms under current master branch).  Of course, mileage may vary but will keep looking at it over the next day.

Reply:
Seems to be working as expected, I've pushed the change in.